### PR TITLE
add MiniMax model client

### PR DESCRIPTION
**Pull request by mxsl-gr**
Hi, this PR is add `MiniMax` model client and has passed unit testing.
I can provide my api_key if needed for testing

the  PR content:

- MiniMax chat client and function calls (not yet supported to return multiple function calls at once)
- MiniMax embedding client
- spring starter
- unit test
- chat client documents
- embedding client documents

`MiniMax` is a popular LLM in China Mainland recently, the link: [MiniMax](https://www.minimaxi.com/)

For some reasons, products from OpenAI and others can't be directly used in Chinese Mainland.

However [ZhiPuAI](https://open.bigmodel.cn/), [Moonshot](https://www.moonshot.cn/), [MiniMax](https://www.minimaxi.com/), [BaiChuanAI](https://www.baichuan-ai.com/) are referred to as the `Four Dragons` of China's LLM.

If they can supported, it will further aid spring-ai to promotion in Chinese Mainland.

If necessary, I can take care of subsequent maintenance since I'm currently using them.

The `Moonshot` PR is #596
The `ZhiPuAI` PR is #623

I'll also provide PR for BaiChuan later.

### CODE REVIEW

**Comment by markpollack** [comment]
Thanks, it has taken a long time for me to get to this but I am reviewing now.  Thanks for such a comprehensive PR!

**Comment by markpollack** [comment]
@mxsl-gr I have sent you an email as I am not able to get an API key on the minimax web site.  I enter my phone number to get a verification code, but the code never comes.

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+		}).toList();
+
+		return new MiniMaxApi.ChatCompletion(chunk.id(), choices, chunk.created(), chunk.model(),
+				chunk.systemFingerprint(), "chat.completion", null, null);
+	}
+
+	/**
+	 * Accessible for testing.
+	 */
+	MiniMaxApi.ChatCompletionRequest createRequest(Prompt prompt, boolean stream) {
```
Comment (markpollack) [inline models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java:224]: In this method, if the model supports media, it needs to be added.  I guess the model does support media which is why the `fromMediaData` method is implemented.  Perhaps it just needs to be called here?

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+ * {@link ChatClient} and {@link StreamingChatClient} implementation for
+ * {@literal MiniMax} backed by {@link MiniMaxApi}.
+ *
+ * @author Geng Rong
+ * @see ChatClient
+ * @see StreamingChatClient
+ * @see MiniMaxApi
+ */
+public class MiniMaxChatClient extends
+		AbstractFunctionCallSupport<MiniMaxApi.ChatCompletionMessage, MiniMaxApi.ChatCompletionRequest, ResponseEntity<MiniMaxApi.ChatCompletion>>
```
Comment (markpollack) [inline models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java:53]: The abstract base class `AbstractFunctionCallSupport` has the method `doChatCompletionStream` that was not implemented.  I'm not sure how you got this code to compile, perhaps there is another commit with that method and some other code as well.  In the meantime, I've added



```
	@Override
	protected Flux<ResponseEntity<MiniMaxApi.ChatCompletion>> doChatCompletionStream(
			MiniMaxApi.ChatCompletionRequest request) {
		throw new RuntimeException("Streaming Function calling is not supported");
	}
```

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java [hunk 3]
```diff
@@ -0,0 +1,890 @@
...
+	 *
+	 * @param embeddingRequest The embedding request.
+	 * @return Returns {@link EmbeddingList}.
+	 * @param <T> Type of the entity in the data list. Can be a {@link String} or {@link List} of tokens (e.g.
+	 * Integers). For embedding multiple inputs in a single request, You can pass a {@link List} of {@link String} or
+	 * {@link List} of {@link List} of tokens. For example:
+	 *
+	 * <pre>{@code List.of("text1", "text2", "text3") or List.of(List.of(1, 2, 3), List.of(3, 4, 5))} </pre>
+	 */
+	public <T> ResponseEntity<EmbeddingList> embeddings(EmbeddingRequest embeddingRequest) {
```
Comment (markpollack) [inline models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java:871]: I believe the `<T>` needs to be removed.

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java [hunk 4]
```diff
@@ -0,0 +1,890 @@
...
+			public static final String AUTO = "auto";
+			/**
+			 * Model will not call a function and instead generates a message
+			 */
+			public static final String NONE = "none";
+
+			/**
+			 * Specifying a particular function forces the model to call that function.
+			 */
+			public static Object FUNCTION(String functionName) {
```
Comment (markpollack) [inline models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java:315]: why is the method name in all capital letters?

**Comment by markpollack** [comment]
I've merged the PR in 6b674014ed942c14a05bf4603eca14fc192466a1  Please review the comments above that were not address in the merge.

Much thanks again

**Comment by mxsl-gr** [comment]
> @mxsl-gr I have sent you an email as I am not able to get an API key on the minimax web site. I enter my phone number to get a verification code, but the code never comes.


@markpollack Regarding the inability to receive verification codes from MiniMax or other model providers, it may be that they are only integrated with SMS providers in mainland China, and phone numbers from other countries or regions cannot receive the messages.

In my email reply, I have provided the test API keys for MiniMax, Moonshot, and DeepSeek.

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java (hunk 1 above)
Comment (mxsl-gr) [reply models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java:224]: MiniMax does not support media. The `fromMediaData` method designed for future model support has been removed in PR #744 .

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java (hunk 2 above)
Comment (mxsl-gr) [reply models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java:53]: resolved in PR #744

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java (hunk 4 above)
Comment (mxsl-gr) [reply models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java:315]: resolved in PR #744

File: models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/common/MiniMaxApiException.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package org.springframework.ai.minimax.api.common;
+
+/**
+ * @author Geng Rong
+ */
+public class MiniMaxApiException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (mxsl-gr) [reply models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/common/MiniMaxApiException.java:6]: resolved in PR #744

**Pull request closed without merging**